python_modules/dagster/dagster/_core/execution/context/compute.py

```diff
@@ -601,25 +600,42 @@ def selected_asset_check_keys(self) -> AbstractSet[AssetCheckKey]:
     @public
     @property
     def selected_output_names(self) -> AbstractSet[str]:
-        """Get the output names that correspond to the current selection of assets this execution is expected to materialize."""
+        """Get the output names that correspond to the current selection of assets and asset checks
+        this execution is expected to materialize.
+        """
         # map selected asset keys to the output names they correspond to
-        selected_asset_keys = self.selected_asset_keys
-        selected_outputs: Set[str] = set()
-        for output_name in self.op.output_dict.keys():
+
+        def _output_is_selected(output_name: str) -> bool:
             asset_info = self.job_def.asset_layer.asset_info_for_output(
                 self.node_handle, output_name
             )
-            if any(  #  For graph-backed assets, check if a downstream asset is selected
+            if asset_info and asset_info.key in self.selected_asset_keys:
+                return True
```

Comment:
For documentation purposes, could we refactor each of these conditionals as separate methods on the `AssetLayer` (`is_selected_asset_output`, `is_selected_graph_backed_asset_output`, `is_selected_asset_check_output`)?

Then an `is_selected_output` method could be created, that is just the `any` of the methods up above.

I prefer this logic to live in the internal machinery.